### PR TITLE
[FSTORE-1306] Use ruff as linter and formatter

### PR DESCRIPTION
**Pull request by vatj**
This PR adds/fixes/changes...
- please summarize your changes to the code 
- and make sure to include all changes to user-facing APIs

JIRA Issue: -

Priority for Review: -

Related PRs: -

**How Has This Been Tested?**

- [ ] Unit Tests
- [ ] Integration Tests
- [ ] Manual Tests on VM

**Checklist For The Assigned Reviewer:**

```
- [ ] Checked if merge conflicts with master exist
- [ ] Checked if stylechecks for Java and Python pass
- [ ] Checked if all docstrings were added and/or updated appropriately
- [ ] Ran spellcheck on docstring
- [ ] Checked if guides & concepts need to be updated
- [ ] Checked if naming conventions for parameters and variables were followed
- [ ] Checked if private methods are properly declared and used
- [ ] Checked if hard-to-understand areas of code are commented
- [ ] Checked if tests are effective
- [ ] Built and deployed changes on dev VM and tested manually
- [x] (Checked if all type annotations were added and/or updated appropriately)
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -50,6 +50,8 @@ coverage.xml
 *.cover
 .hypothesis/
 .pytest_cache/
+.ruff_cache/
+bigquery.json
```
Comment (dhananjay-mk) [inline .gitignore:54]: the bigquery.json should be removed I guess?

File: python/hsfs/client/base.py [hunk 2]
```diff
@@ -17,16 +17,16 @@
...
 
+import furl
 import requests
 import urllib3
-
-from hsfs.client import exceptions, auth
+from hsfs.client import auth, exceptions
 from hsfs.decorators import connected
 
+
```
Comment (dhananjay-mk) [inline python/hsfs/client/base.py:29]: are these additional empty lines being introduced by ruff? I guess yes

Comment (vatj) [reply python/hsfs/client/base.py:29]: yes, double empty lines between import other stuff

File: .gitignore (hunk 1 above)
Comment (vatj) [reply .gitignore:54]: The file is not there in this PR and it should be part of gitignore

**Review by robzor92 — CHANGES_REQUESTED** [review]
If we find issues with the PR I will very likely be reverting it. I have gone through it and did not find an obvious issue but there are many code paths and integrations which we don't test in the nightly tests. So a green build does not mean we are out of the woods yet here.

**Review by robzor92 — APPROVED** [review]

**Pull request merged**